buildscript/sa.go

```diff
@@ -48,3 +49,26 @@ func SAModTidy(ctx *task.Context) error {
 
 	return nil
 }
+
+// SAEvergreenValidate runs `evergreen validate` on common.yml and ensures the file is valid.
+func SAEvergreenValidate(ctx *task.Context) error {
+	output, err := sh.RunOutput(ctx, "evergreen", "validate", "--file", "common.yml")
+	if err != nil {
+		return fmt.Errorf("error from `evergreen validate`: %s: %w", output, err)
+	}
+
+	// TODO: change this if-block in TOOLS-...
```

Comment:
Since there's some shortcomings with the parser detailed in [EVG-14227](https://jira.mongodb.org/browse/EVG-14227), I'm thinking of making a TOOLS ticket blocked on that EVG ticket for removing the duplicate key special-casing below (I'll add it once everything is lgtm'd).

The special-casing is just looping through the `evergreen validate` output and only returns an error if it sees a warning that's not associated with duplicate keys. It shouldn't break when the Evergreen team updates to `yaml.v3`, since these types of warnings will just disappear.

Reply:
> I'm thinking of making a TOOLS ticket blocked on that EVG ticket for removing the duplicate key special-casing below

I think this is a good idea. I'd say just go ahead and file it now; no need to wait for this to get LGTM'ed